hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -281,18 +277,24 @@ private Diagnostics initDiagnostics() {
         return new Diagnostics(name, logger, instanceName, properties);
     }
 
-    private MetricsRegistryImpl initMetricsRegistry() {
-        ProbeLevel probeLevel = properties.getEnum(Diagnostics.METRICS_LEVEL, ProbeLevel.class);
-        ILogger logger = loggingService.getLogger(MetricsRegistryImpl.class);
-        MetricsRegistryImpl metricsRegistry = new MetricsRegistryImpl(getName(), logger, probeLevel);
-        RuntimeMetricSet.register(metricsRegistry);
-        GarbageCollectionMetricSet.register(metricsRegistry);
-        OperatingSystemMetricSet.register(metricsRegistry);
-        ThreadMetricSet.register(metricsRegistry);
-        ClassLoadingMetricSet.register(metricsRegistry);
-        FileMetricSet.register(metricsRegistry);
-        metricsRegistry.scanAndRegister(clientExtension.getMemoryStats(), "memory");
-        return metricsRegistry;
+    private void initProbeSources() {
+        probeRegistry.register(Probing.GC);
```

Comment:
I would like to see see Probe.GC, Probe.OS moved to a seperate class. Should not be part of 'Probing' which sounds very generic.

Reply:
You are too early ;) PR not ready. And they are in a separate class. I just want that class to not be public to avoid lots of classes in the wild so I put a constant in the util.